api/python/quilt3/telemetry.py

```diff
@@ -0,0 +1,146 @@
+import atexit
+from concurrent.futures import ThreadPoolExecutor, wait
+import functools
+import os
+import platform
+from requests_futures.sessions import FuturesSession
+import sys
+from threading import Lock
+import uuid
+
+from .util import load_config, set_config_value
+from . import __version__ as quilt3_version
+
+TELEMETRY_URL = "https://telemetry.quiltdata.cloud/Prod/metrics"
+TELEMETRY_USER_AGENT = "QuiltCli"
+TELEMETRY_CLIENT_TYPE = "quilt3-python-client"
+TELEMETRY_SCHEMA_VERSION = "pyclient-usage-metrics-v0"
+
+DISABLE_USAGE_METRICS_ENVVAR = "QUILT_DISABLE_USAGE_METRICS"
+MAX_CLEANUP_WAIT_SECS = 5
+
+
+
+
+class ApiTelemetry:
+    session = None
+    session_id = str(uuid.uuid4())
+    pending_reqs = []
+    pending_reqs_lock = Lock()
+    telemetry_disabled = False
+
+    @staticmethod
```

Comment:
Use `@classmethod` for everything; the existence of `@staticmethod` is kind of a mistake in Python.

Reply:
I don't think that's what we want to do. You should not be able to call these methods from an instance of the class.

Reply:
Sure... But `@staticmethod` doesn't stop you from doing that:

```
In [1]: class Foo():
   ...:     @staticmethod
   ...:     def blah():
   ...:         print('blah')
   ...:

In [2]: Foo.blah()
blah

In [3]: x = Foo()

In [4]: x.blah()
blah
```

Basically, it has no redeeming qualities. Also, according to Guido:
"""
They're basically an accident -- back in the Python 2.2 days when I was inventing new-style classes and descriptors, I meant to implement class methods but at first I didn't understand them and accidentally implemented static methods first. Then it was too late to remove them and only provide class methods.
"""